app/services/upload/get_import_animals.go

```diff
@@ -0,0 +1,67 @@
+package upload
+
+import (
+	"errors"
+	"fmt"
+
+	"github.com/goravel/framework/facades"
+	"github.com/startup-of-zero-reais/zoo-api/app/helpers"
+	"github.com/startup-of-zero-reais/zoo-api/app/http/responses"
+	"github.com/startup-of-zero-reais/zoo-api/app/models"
+)
+
+func (e *uploadImpl) GetImportAnimals(ids []string) ([]models.Animal, error) {
+	var importAnimals []models.ImportAnimals
+
+	err := facades.Orm().Query().Where("id IN ?", ids).Find(&importAnimals)
+	if err != nil {
+		facades.Log().Errorf("failed to get import animals %v", err)
+		return nil, responses.ErrUnhandledPgError
+	}
+
+	missingIDs := helpers.FindMissingIDs(ids, importAnimals, func(ia models.ImportAnimals) string {
+		return ia.ID
+	})
```

Comment:
Se o helper de `FindMissingIDs` for mais genérico, você pode usar aqui, de forma mais clara:

```go
missingIDs := helpers.Filter(importAnimals, func (i int, ia models.ImportAnimals) bool {
  for _, id := range ids {
    if id == ia.ID { // se houver algum importAnimals com esse id, devemos filtrá-lo
      return false // false para filtrar
    }
  }

  return true // se o loop finalizou e não encontrou o id, ele é missing.
})
```

Pra mim dessa forma fica menos abstrato o que o `FindMissingIDs` faz. Apesar do nome ser bem declarativo, não sei como ele trabalha.